exporter/splunkexporter/exporter.go

```diff
@@ -0,0 +1,86 @@
+package splunkexporter
+
+import (
+	"compress/gzip"
+	"context"
+	"errors"
+	"fmt"
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/consumer/consumerdata"
+	"go.opentelemetry.io/collector/consumer/pdatautil"
+	"go.opentelemetry.io/collector/obsreport"
+	"go.uber.org/zap"
+	"net/http"
+	"net/url"
+	"sync"
+	"time"
+)
+
+type splunkExporter struct {
+	pushMetricsData func(ctx context.Context, md consumerdata.MetricsData) (droppedTimeSeries int, err error)
+}
+
+type exporterOptions struct {
+  url   *url.URL
+  token string
+}
+
+
+// New returns a new Splunk exporter.
+func New(
+	config *Config,
+	logger *zap.Logger,
+) (component.MetricsExporterOld, error) {
+
+	if config == nil {
+		return nil, errors.New("nil config")
+	}
+
+	options, err := config.getOptionsFromConfig()
+	if err != nil {
+		return nil,
+			fmt.Errorf("failed to process %q config: %v", config.Name(), err)
+	}
+
+	logger.Info("Splunk Config", zap.String("url", options.url.String()))
+
+	if config.Name() == "" {
+		config.SetType(typeStr)
+		config.SetName(typeStr)
+	}
+
+	dpClient := &client{
+		url: options.url,
+		client: &http.Client{
+			// TODO: What other settings of http.Client to expose via config?
+			//  Or what others change from default values?
+			Timeout: 5 * time.Second,
```

Comment:
See `sapmexporter` as an example of exporter that would be good to mimic. We typically expose compression option, number of concurrent connections (if you plan to add support for concurrent connections to this exporter).

Reply:
I have taken what I could for now, but definitely more could be done with workers to send parallel payloads.